### PR TITLE
HIVE-15826: Add 'serialization.encoding' To All SerDes

### PR DESCRIPTION
**Pull request by guptashailesh92**
Added support for **serialization.encoding** for OpenCSV, JsonSerde and RegexSerde.

Issue: [HIVE-15826](https://issues.apache.org/jira/browse/HIVE-15826)

### CODE REVIEW

**Comment by guptashailesh92** [comment]
@kasakrisz @abstractdog Can you please review? The qtests test failure seems unrelated. If thats not the case, please let me know.

**Review by kasakrisz — APPROVED** [review]
LGTM, pending tests

File: ql/src/test/queries/clientpositive/serde_opencsv.q [hunk 1]
```diff
@@ -28,9 +29,10 @@ ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.OpenCSVSerde'
...
   "quoteChar"     = "\'",
-  "escapeChar"    = "\\"
+  "escapeChar"    = "\\",
+  "serialization.encoding" = "ISO8859_1"
 ) stored as textfile;
 
 LOAD DATA LOCAL INPATH "../../data/files/opencsv-data.txt" INTO TABLE serde_opencsv;
 
-SELECT count(*) FROM serde_opencsv;
+SELECT * FROM serde_opencsv;
```
Comment (abstractdog) [inline ql/src/test/queries/clientpositive/serde_opencsv.q:38]: was there any purpose of the original count(*) query? without knowing the history, I believe we can leave both queries here

Comment (guptashailesh92) [reply ql/src/test/queries/clientpositive/serde_opencsv.q:40]: thank you for the review. will update.

**Review by abstractdog — APPROVED** [review]
+1

**Pull request merged**